types/yeoman-test/index.d.ts

```diff
@@ -217,6 +225,145 @@ export interface RunResult extends RunResultOptions {
      * Dumps the name of each file to the console.
      */
     dumpFilenames(): void;
+
+    /**
+     * Assert that a file exists
+     * @param path     - path to a file
+     * @example
+     * result.assertFile('templates/user.hbs');
+     *
+     * @also
+     *
+     * Assert that each files in the array exists
+     * @param paths    - an array of paths to files
+     * @example
+     * result.assertFile(['templates/user.hbs', 'templates/user/edit.hbs']);
+     */
+    assertFile(path: string | string[]): void;
+
+    /**
+     * Assert that a file doesn't exist
+     * @param file     - path to a file
+     * @example
+     * result.assertNoFile('templates/user.hbs');
+     *
+     * @also
+     *
+     * Assert that each of an array of files doesn't exist
+     * @param pairs    - an array of paths to files
+     * @example
+     * result.assertNoFile(['templates/user.hbs', 'templates/user/edit.hbs']);
+     */
+    assertNoFile(file: string | string[]): void;
+
+    /**
+     * Assert that a file's content matches a regex or string
+     * @param file     - path to a file
+     * @param reg      - regex / string that will be used to search the file
+     * @example
+     * result.assertFileContent('models/user.js', /App\.User = DS\.Model\.extend/);
+     * result.assertFileContent('models/user.js', 'App.User = DS.Model.extend');
+     *
+     * @also
+     *
+     * Assert that each file in an array of file-regex pairs matches its corresponding regex
+     * @param pairs    - an array of arrays, where each subarray is a [String, RegExp] pair
+     * @example
+     * var arg = [
+     *   [ 'models/user.js', /App\.User = DS\.Model\.extend/ ],
+     *   [ 'controllers/user.js', /App\.UserController = Ember\.ObjectController\.extend/ ]
+     * ]
+     * result.assertFileContent(arg);
+     */
+    assertFileContent(file: string, reg: string | RegExp): void;
+    assertFileContent(pairs: Array<[string, RegExp]>): void;
```

Comment:
The second item of the `pairs`-tubles can be `string` as well (=> `Array<[string, string | RegExp]>`)